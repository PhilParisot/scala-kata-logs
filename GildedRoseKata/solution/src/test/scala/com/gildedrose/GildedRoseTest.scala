package com.gildedrose

import org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations
import org.approvaltests.core.ApprovalFailureReporter
import org.approvaltests.reporters.{QuietReporter, UseReporter}
import org.junit.jupiter.api.Test

@UseReporter(Array[ApprovalFailureReporter](classOf[QuietReporter]))
class GildedRoseTest {
  @Test def update_quality_of_items: Unit = {
    verifyAllCombinations(
      (name, sellIn, quantity) => updateQuality(name, sellIn, quantity),
      Array[String](
        "a common item",
        "Aged Brie",
        "Backstage passes to a TAFKAL80ETC concert",
        "Sulfuras, Hand of Ragnaros"
      ),
      Array[Integer](-100, -1, 0, 1, 2, 6, 8, 11),
      Array[Integer](-1, 0, 1, 49, 50)
    )
  }

  private def updateQuality(
      name: String,
      sellIn: Integer,
      quality: Integer
  ): String = {
    val items = List(new Item(name, sellIn, quality))
    val gildedRose = new GildedRose(items.toArray)
    gildedRose.updateQuality()
    gildedRose.items(0).toString
  }
}
