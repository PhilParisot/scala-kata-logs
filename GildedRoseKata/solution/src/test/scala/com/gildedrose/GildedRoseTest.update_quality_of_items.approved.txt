[a common item, -100, -1] => a common item, -101, -1 
[a common item, -100, 0] => a common item, -101, 0 
[a common item, -100, 1] => a common item, -101, 0 
[a common item, -100, 49] => a common item, -101, 47 
[a common item, -100, 50] => a common item, -101, 48 
[a common item, -1, -1] => a common item, -2, -1 
[a common item, -1, 0] => a common item, -2, 0 
[a common item, -1, 1] => a common item, -2, 0 
[a common item, -1, 49] => a common item, -2, 47 
[a common item, -1, 50] => a common item, -2, 48 
[a common item, 0, -1] => a common item, -1, -1 
[a common item, 0, 0] => a common item, -1, 0 
[a common item, 0, 1] => a common item, -1, 0 
[a common item, 0, 49] => a common item, -1, 47 
[a common item, 0, 50] => a common item, -1, 48 
[a common item, 1, -1] => a common item, 0, -1 
[a common item, 1, 0] => a common item, 0, 0 
[a common item, 1, 1] => a common item, 0, 0 
[a common item, 1, 49] => a common item, 0, 48 
[a common item, 1, 50] => a common item, 0, 49 
[a common item, 2, -1] => a common item, 1, -1 
[a common item, 2, 0] => a common item, 1, 0 
[a common item, 2, 1] => a common item, 1, 0 
[a common item, 2, 49] => a common item, 1, 48 
[a common item, 2, 50] => a common item, 1, 49 
[a common item, 6, -1] => a common item, 5, -1 
[a common item, 6, 0] => a common item, 5, 0 
[a common item, 6, 1] => a common item, 5, 0 
[a common item, 6, 49] => a common item, 5, 48 
[a common item, 6, 50] => a common item, 5, 49 
[a common item, 8, -1] => a common item, 7, -1 
[a common item, 8, 0] => a common item, 7, 0 
[a common item, 8, 1] => a common item, 7, 0 
[a common item, 8, 49] => a common item, 7, 48 
[a common item, 8, 50] => a common item, 7, 49 
[a common item, 11, -1] => a common item, 10, -1 
[a common item, 11, 0] => a common item, 10, 0 
[a common item, 11, 1] => a common item, 10, 0 
[a common item, 11, 49] => a common item, 10, 48 
[a common item, 11, 50] => a common item, 10, 49 
[Aged Brie, -100, -1] => Aged Brie, -101, 1 
[Aged Brie, -100, 0] => Aged Brie, -101, 2 
[Aged Brie, -100, 1] => Aged Brie, -101, 3 
[Aged Brie, -100, 49] => Aged Brie, -101, 50 
[Aged Brie, -100, 50] => Aged Brie, -101, 50 
[Aged Brie, -1, -1] => Aged Brie, -2, 1 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 1, -1] => Aged Brie, 0, 0 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 1] => Aged Brie, 0, 2 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 2, -1] => Aged Brie, 1, 0 
[Aged Brie, 2, 0] => Aged Brie, 1, 1 
[Aged Brie, 2, 1] => Aged Brie, 1, 2 
[Aged Brie, 2, 49] => Aged Brie, 1, 50 
[Aged Brie, 2, 50] => Aged Brie, 1, 50 
[Aged Brie, 6, -1] => Aged Brie, 5, 0 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 8, -1] => Aged Brie, 7, 0 
[Aged Brie, 8, 0] => Aged Brie, 7, 1 
[Aged Brie, 8, 1] => Aged Brie, 7, 2 
[Aged Brie, 8, 49] => Aged Brie, 7, 50 
[Aged Brie, 8, 50] => Aged Brie, 7, 50 
[Aged Brie, 11, -1] => Aged Brie, 10, 0 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -100, -1] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 0] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 1] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 49] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -100, 50] => Backstage passes to a TAFKAL80ETC concert, -101, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, -1] => Backstage passes to a TAFKAL80ETC concert, 0, 2 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert, 0, 4 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 2, -1] => Backstage passes to a TAFKAL80ETC concert, 1, 2 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Backstage passes to a TAFKAL80ETC concert, 1, 3 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Backstage passes to a TAFKAL80ETC concert, 1, 4 
[Backstage passes to a TAFKAL80ETC concert, 2, 49] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 2, 50] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, -1] => Backstage passes to a TAFKAL80ETC concert, 5, 1 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 8, -1] => Backstage passes to a TAFKAL80ETC concert, 7, 1 
[Backstage passes to a TAFKAL80ETC concert, 8, 0] => Backstage passes to a TAFKAL80ETC concert, 7, 2 
[Backstage passes to a TAFKAL80ETC concert, 8, 1] => Backstage passes to a TAFKAL80ETC concert, 7, 3 
[Backstage passes to a TAFKAL80ETC concert, 8, 49] => Backstage passes to a TAFKAL80ETC concert, 7, 50 
[Backstage passes to a TAFKAL80ETC concert, 8, 50] => Backstage passes to a TAFKAL80ETC concert, 7, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, -1] => Backstage passes to a TAFKAL80ETC concert, 10, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, -100, -1] => Sulfuras, Hand of Ragnaros, -100, -1 
[Sulfuras, Hand of Ragnaros, -100, 0] => Sulfuras, Hand of Ragnaros, -100, 0 
[Sulfuras, Hand of Ragnaros, -100, 1] => Sulfuras, Hand of Ragnaros, -100, 1 
[Sulfuras, Hand of Ragnaros, -100, 49] => Sulfuras, Hand of Ragnaros, -100, 49 
[Sulfuras, Hand of Ragnaros, -100, 50] => Sulfuras, Hand of Ragnaros, -100, 50 
[Sulfuras, Hand of Ragnaros, -1, -1] => Sulfuras, Hand of Ragnaros, -1, -1 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 1, -1] => Sulfuras, Hand of Ragnaros, 1, -1 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 1] => Sulfuras, Hand of Ragnaros, 1, 1 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 2, -1] => Sulfuras, Hand of Ragnaros, 2, -1 
[Sulfuras, Hand of Ragnaros, 2, 0] => Sulfuras, Hand of Ragnaros, 2, 0 
[Sulfuras, Hand of Ragnaros, 2, 1] => Sulfuras, Hand of Ragnaros, 2, 1 
[Sulfuras, Hand of Ragnaros, 2, 49] => Sulfuras, Hand of Ragnaros, 2, 49 
[Sulfuras, Hand of Ragnaros, 2, 50] => Sulfuras, Hand of Ragnaros, 2, 50 
[Sulfuras, Hand of Ragnaros, 6, -1] => Sulfuras, Hand of Ragnaros, 6, -1 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 1 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 8, -1] => Sulfuras, Hand of Ragnaros, 8, -1 
[Sulfuras, Hand of Ragnaros, 8, 0] => Sulfuras, Hand of Ragnaros, 8, 0 
[Sulfuras, Hand of Ragnaros, 8, 1] => Sulfuras, Hand of Ragnaros, 8, 1 
[Sulfuras, Hand of Ragnaros, 8, 49] => Sulfuras, Hand of Ragnaros, 8, 49 
[Sulfuras, Hand of Ragnaros, 8, 50] => Sulfuras, Hand of Ragnaros, 8, 50 
[Sulfuras, Hand of Ragnaros, 11, -1] => Sulfuras, Hand of Ragnaros, 11, -1 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
